src/sheldon_suggestions_server.erl

```diff
@@ -0,0 +1,70 @@
+-module(sheldon_suggestions_server).
+
+-behaviour(gen_server).
+
+%%% API
+-export([ suggestions/2
+        , add_suggestions/2
+        ]).
+
+%%% gen_server callbacks
+-export([
+  init/1,
+  handle_call/3,
+  handle_cast/2,
+  handle_info/2,
+  terminate/2,
+  code_change/3
+]).
+
+%%%=============================================================================
+%%% API
+%%%=============================================================================
+
+-spec suggestions([sheldon_result:misspelled_word()], sheldon_dictionary:language()) ->
+  [sheldon_result:misspelled_word()].
+suggestions(MisspelledWords, Lang) ->
+  wpool:call(suggestions_pool, {suggest, MisspelledWords, Lang}, available_worker, infinity).
+
+-spec add_suggestions(sheldon_result:misspelled_word(), sheldon_dictionary:language()) ->
+  sheldon_result:misspelled_word().
+add_suggestions(#{word := Word} = MisspelledWord, Lang) ->
+  Candidates = sheldon_dictionary:candidates(Word, Lang),
+  MisspelledWord#{candidates => Candidates}.
+
+%%%=============================================================================
+%%% gen_server callbacks
+%%%=============================================================================
+
+-spec init(no_args) -> {ok, map()}.
+init(no_args) ->
+  {ok, #{}}.
+
+-spec handle_call(term(), term(), map()) -> {reply, ok, map()}.
+handle_call({suggest, MisspelledWords, Lang}, _From, State) ->
+  Reply = try rpc:pmap({?MODULE, add_suggestions}, [Lang], MisspelledWords) of
```

Comment:
This shouldn't be done in the worker context. To actually parallelize the task, this should be performed in the _client_ side (i.e. within _suggestions/2_) and the mapped function should be the one invoking `wpool:call(…)`.